data/net/minecraft/world/entity/npc/VillagerTrades.mapping

```diff
@@ -1,47 +1,145 @@
 CLASS net/minecraft/world/entity/npc/VillagerTrades
-	CLASS DyedArmorForEmeralds
+	METHOD toIntMap (Lcom/google/common/collect/ImmutableMap;)Lit/unimi/dsi/fastutil/ints/Int2ObjectMap;
+		ARG 0 map
+	CLASS ItemListing
 		METHOD getOffer (Lnet/minecraft/world/entity/Entity;Ljava/util/Random;)Lnet/minecraft/world/item/trading/MerchantOffer;
 			ARG 1 trader
 			ARG 2 rand
 	CLASS EmeraldForItems
+		METHOD <init> (Lnet/minecraft/world/level/ItemLike;III)V
+			ARG 1 item
+			ARG 2 cost
+			ARG 3 maxUses
+			ARG 4 villagerXp
 		METHOD getOffer (Lnet/minecraft/world/entity/Entity;Ljava/util/Random;)Lnet/minecraft/world/item/trading/MerchantOffer;
 			ARG 1 trader
 			ARG 2 rand
-	CLASS EmeraldsForVillagerTypeItem
+	CLASS ItemsForEmeralds
+		METHOD <init> (Lnet/minecraft/world/item/Item;III)V
+			ARG 1 item
+			ARG 2 emeraldCost
+			ARG 3 numberOfItems
+			ARG 4 villagerXp
+		METHOD <init> (Lnet/minecraft/world/item/Item;IIII)V
+			ARG 1 item
+			ARG 2 emeraldCost
+			ARG 3 numberOfItems
+			ARG 4 maxUses
+			ARG 5 villagerXp
+		METHOD <init> (Lnet/minecraft/world/item/ItemStack;IIII)V
+			ARG 1 itemStack
+			ARG 2 emeraldCost
+			ARG 3 numberOfItems
+			ARG 4 maxUses
+			ARG 5 villagerXp
+		METHOD <init> (Lnet/minecraft/world/item/ItemStack;IIIIF)V
+			ARG 1 itemStack
+			ARG 2 emeralCost
+			ARG 3 numberOfItems
+			ARG 4 maxUses
+			ARG 5 villagerXp
+			ARG 6 priceMultiplier
+		METHOD <init> (Lnet/minecraft/world/level/block/Block;IIII)V
+			ARG 1 block
+			ARG 2 emeraldCost
+			ARG 3 numberOfItems
+			ARG 4 maxUses
+			ARG 5 villagerXp
 		METHOD getOffer (Lnet/minecraft/world/entity/Entity;Ljava/util/Random;)Lnet/minecraft/world/item/trading/MerchantOffer;
 			ARG 1 trader
 			ARG 2 rand
-	CLASS EnchantBookForEmeralds
+	CLASS DyedArmorForEmeralds
+		METHOD <init> (Lnet/minecraft/world/item/Item;I)V
+			ARG 1 item
+			ARG 2 value
+		METHOD <init> (Lnet/minecraft/world/item/Item;III)V
+			ARG 1 item
+			ARG 2 value
+			ARG 3 maxUses
+			ARG 4 villagerXp
 		METHOD getOffer (Lnet/minecraft/world/entity/Entity;Ljava/util/Random;)Lnet/minecraft/world/item/trading/MerchantOffer;
 			ARG 1 trader
 			ARG 2 rand
-	CLASS EnchantedItemForEmeralds
+		METHOD getRandomDye (Ljava/util/Random;)Lnet/minecraft/world/item/DyeItem;
+			ARG 0 random
+	CLASS EnchantBookForEmeralds
+		METHOD <init> (I)V
+			ARG 1 villagerXp
 		METHOD getOffer (Lnet/minecraft/world/entity/Entity;Ljava/util/Random;)Lnet/minecraft/world/item/trading/MerchantOffer;
 			ARG 1 trader
 			ARG 2 rand
-	CLASS ItemListing
```

Comment:
Why did this get removed?

Reply:
Seems it got rearranged to up here:
https://github.com/ParchmentMC/Parchment/pull/18/files/4c1b1d3abf611a0e837198fff444e171d8f0abbf#diff-dc2071c5ff041b39b463bc14e09b4aab0eab2e2a06477a79400a43cd367de3a3R4